### PR TITLE
Removing the deprecated methods and renaming the ones those have the …

### PR DESCRIPTION
**Pull request by RenatoChitolina**
# Summary
### Removed
- Removed the deprecated `registerOffchain` method
- Removed the deprecated `isRegisteredOnchain` method
- Removed the deprecated `transfer` method
- Removed the deprecated `batchNftTransfer` method
- Removed the deprecated `burn` method
- Removed the deprecated `prepareWithdrawal` method
- Removed the deprecated `createOrder` method
- Removed the deprecated `cancelOrder` method
- Removed the deprecated `createTrade` method

### Changed
- Renamed the `registerOffchainWithSigner` method to `registerOffchain`
- Renamed the `isRegisteredOnchainWithSigner` method to `isRegisteredOnchain`
- Renamed the `transferWithSigner` method to `transfer`
- Renamed the `batchNftTransferWithSigner` method to `batchNftTransfer`
- Renamed the `burnWithSigner` method to `burn`
- Renamed the `prepareWithdrawalWithSigner` method to `prepareWithdrawal`
- Renamed the `createOrderWithSigner` method to `createOrder`
- Renamed the `cancelOrderWithSigner` method to `cancelOrder`
- Renamed the `createTradeWithSigner` method to `createTrade`

# Why the changes
This is the last step to accomplish the Core SDK compatibility adaptation to the Wallets SDK

# Things worth calling out
Equivalent PR on **imx-testing**: https://github.com/immutable/imx-testing/pull/82

### CODE REVIEW

**Review by carmen0208 — COMMENTED** [review]
We probably need  to change the README as well because for some example, get StarkWallet is not making sense anymore

**Review by ABashkirtseva — APPROVED** [review]

**Comment by RenatoChitolina** [comment]
> We probably need to change the README as well because for some example, get StarkWallet is not making sense anymore

Yes, I'm updating it to introduce the Wallets SDK and how it is used to provide what Core SDK needs

**Review by Kastet — CHANGES_REQUESTED** [review]
@RenatoChitolina nice work!

Now is good opportunity to clean other stuff up
- from the `/utils` module, things like `serializeSignature`, `sign`, `signRaw` and anything that's no longer used needs to be removed or at the very least made private (removed from the SDK public interface)
- is `@ethersproject/abstract-signer` still used?
- there are pretty useless comments in the workflows, like `// Sign message with L1 credentials` and `// Obtain Eth address from signer`, etc. that state the obvious  - I know they carried over from the existing implementation but it doesn't mean this should remain, let's remove them

**Comment by RenatoChitolina** [comment]
> @RenatoChitolina nice work!
> 
> Now is good opportunity to clean other up stuff up
> 
>     * from the `/utils` module, things like `serializeSignature`, `sign`, `signRaw` and anything that's no longer used needs to be removed or at the very least made private (removed from the SDK public interface)
> 
>     * is `@ethersproject/abstract-signer` still used?
> 
>     * there are pretty useless comments in the workflows, like `// Sign message with L1 credentials` and `// Obtain Eth address from signer`, etc. that state the obvious  - I know they carried over from the existing implementation but it doesn't mean this should remain, let's remove them

@Kastet It was manageable for now to remove the `serializeSignature` and `sign`. The `signRaw` is still used on Mint, for example. The @ethersproject/abstract-signer provides the L1Signer interface, so It still remains as well. Regarding comments, I've removed some of those I considered not necessary.

File: README.md [hunk 1]
```diff
@@ -58,32 +58,33 @@ const ethNetwork = 'ropsten'; // or mainnet;
...
 const privateKey = YOUR_PRIVATE_KEY;
 const provider = new AlchemyProvider(ethNetwork, YOUR_ALCHEMY_API_KEY);
 const signer = new Wallet(privateKey).connect(provider);
 ```
 
-#### Stark Wallet
+#### WalletConnection
 
-Some methods require a stark wallet as a parameter. The Core SDK expects you will generate your own stark wallet.
+WalletConnection is a top level connector which contains all the needed resources (eg.: signers for L1 and L2) to possibilitate Core SDK workflows usage.
```
Comment (Kastet) [inline README.md:69]: @sprngbk can you please help polish the wording here 🙏

File: README.md [hunk 2]
```diff
@@ -203,23 +204,35 @@ A workflow is a combination of API and contract calls required for more complica
...
+const privateKey = 'UPDATE WITH THE PRIVATE KEY HERE';
+const l1Wallet = new Wallet(privateKey);
+const l1Signer = l1Wallet.connect(alchemyProvider);
 
-const registerUser = async () => {
-  const response = await workflows.registerOffchain(signer);
-  console.log(response);
-};
+// Sets up the L2Signer
+const l2Wallet = await generateStarkWallet(l1Signer);
```
Comment (Kastet) [inline README.md:226]: @RenatoChitolina is that intentionally called `l2Wallet`? Should this be just `l2Signer` to be consistent with the rest of the code?

File: README.md (hunk 1 above)
Comment (RenatoChitolina) [reply README.md:69]: > "WalletConnection is a high-level connector that contains all necessary resources (eg.: signers for L1 and L2) that make it possible to operate through the Core SDK workflows."

@Kastet  Wdyt?

**Review by Kastet — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -66,32 +66,33 @@ const ethNetwork = 'ropsten'; // or mainnet;
...
 const privateKey = YOUR_PRIVATE_KEY;
 const provider = new AlchemyProvider(ethNetwork, YOUR_ALCHEMY_API_KEY);
 const signer = new Wallet(privateKey).connect(provider);
 ```
 
-#### Stark Wallet
+#### WalletConnection
 
-Some methods require a stark wallet as a parameter. The Core SDK expects you will generate your own stark wallet.
+WalletConnection is a top level connector which contains all the needed resources (eg.: signers for L1 and L2) to possibilitate Core SDK workflows usage.
```
Comment (sprngbk) [inline README.md:77]:
```suggestion
WalletConnection is a top level connector which contains all the required resources (eg. L1 and L2 signers) to enable usage of the Core SDK workflows.
```

**Review by sprngbk — APPROVED** [review]

**Pull request merged**